File: backend/Dockerfiles/Dockerfile.swift
# syntax=docker/dockerfile:1

FROM dhi.io/alpine-base:3.22-dev AS common

# We're using Alpine's /bin/sh, so disable pipefail suggestion.
# hadolint global ignore=DL4006

ARG SWIFTLINT_VER
ARG SWIFTLINT_SHA

RUN apk add --no-cache curl ca-certificates && \
    curl -fsSL -o /tmp/swiftlint.zip "https://github.com/realm/SwiftLint/releases/download/${SWIFTLINT_VER}/swiftlint_linux.zip" && \
    echo "$SWIFTLINT_SHA  /tmp/swiftlint.zip" | sha256sum -c - && \
    unzip /tmp/swiftlint.zip -d /opt/swiftlint && \
    chmod a+x /opt/swiftlint/swiftlint && \
    apk del curl


FROM swift:5.5.2-focal

ARG MAINTAINER
LABEL maintainer=$MAINTAINER

# Install swiftlint
COPY --from=common /opt/swiftlint/ /opt/swiftlint/
COPY ./engine/plugins/swiftlint/swiftlint.yml /etc/swiftlint.yml

ENV PATH=/opt/swiftlint:$PATH

File: backend/Dockerfiles/Dockerfile.php
# Build stage with dev tools
FROM dhi.io/composer:2.9-debian13-php8.4-dev AS builder

ARG PHP_SCANNER_VER

# Install dependencies for downloading tools
# `--fix-broken` step is added to fix a (probably temporary) regression in the base image. If
#   you're seeing this in the future, they may have fixed this, so feel free to remove that line
#   and try again
RUN apt-get update && \
    apt-get --fix-broken -y install && \
    apt-get install -y --no-install-recommends curl && \
    rm -rf /var/lib/apt/lists/*

WORKDIR /tmp

# Download Psalm
RUN curl -fsSL https://github.com/vimeo/psalm/releases/download/${PHP_SCANNER_VER}/psalm.phar -o /tmp/psalm && \
    chmod a+x /tmp/psalm

# Final runtime stage - clean image without dev packages
FROM dhi.io/composer:2.9-debian13-php8.4-dev

ARG MAINTAINER
LABEL maintainer=$MAINTAINER

# Copy tools from builder stage
COPY --from=builder /tmp/psalm /usr/local/bin/psalm

ENTRYPOINT [ "" ]
